cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/dataset/MetadataDatasetTest.java

```diff
@@ -201,26 +201,38 @@ public void testSearchOnTags() throws Exception {
     Assert.assertEquals(0, dataset.getTags(stream1).size());
     dataset.addTags(app1, "tag1", "tag2", "tag3");
     dataset.addTags(flow1, "tag1");
-    dataset.addTags(dataset1, "tag3", "tag2", "tag12");
+    dataset.addTags(dataset1, "tag3", "tag2", "tag12-tag33");
```

Comment:
I agree we shouldn't do any splitting for tags besides on tag separator to get individual tags.
@poornachandra : Thoughts ?